qiskit/algorithms/evolvers/trotterization/trotter_qrte.py

```diff
@@ -0,0 +1,245 @@
+from qiskit.opflow import (
+    SummedOp,
+    PauliOp,
+    CircuitOp,
+    ExpectationBase,
+    CircuitSampler,
+    PauliSumOp,
+    StateFn,
+    OperatorBase,
+)
+from qiskit.circuit.library import PauliEvolutionGate
+from qiskit.providers import Backend
+from qiskit.synthesis import ProductFormula, LieTrotter
+from qiskit.utils import QuantumInstance
+
+
+class TrotterQRTE(RealEvolver):
+    """Quantum Real Time Evolution using Trotterization.
+    Type of Trotterization is defined by a ProductFormula provided.
+
+    Examples:
+
+        .. jupyter-execute::
+
+            from qiskit.opflow import X, Z, Zero
+            from qiskit.algorithms import EvolutionProblem, TrotterQRTE
+            from qiskit import BasicAer
+            from qiskit.utils import QuantumInstance
+
+            operator = X + Z
+            initial_state = Zero
+            time = 1
+            evolution_problem = EvolutionProblem(operator, 1, initial_state)
+            # LieTrotter with 1 rep
+            backend = BasicAer.get_backend("statevector_simulator")
+            quantum_instance = QuantumInstance(backend=backend)
+            trotter_qrte = TrotterQRTE(quantum_instance=quantum_instance)
+            evolved_state = trotter_qrte.evolve(evolution_problem).evolved_state
+    """
+
+    def __init__(
+        self,
+        product_formula: Optional[ProductFormula] = None,
+        expectation: Optional[ExpectationBase] = None,
+        quantum_instance: Optional[Union[QuantumInstance, Backend]] = None,
+    ) -> None:
+        """
+        Args:
+            product_formula: A Lie-Trotter-Suzuki product formula. The default is the Lie-Trotter
+                first order product formula with a single repetition.
+            expectation: An instance of ExpectationBase which defines a method for calculating
+                expectation values of EvolutionProblem.aux_operators.
+            quantum_instance: A quantum instance used for calculating expectation values of
+                EvolutionProblem.aux_operators.
+        """
+        if product_formula is None:
+            product_formula = LieTrotter()
+        self._product_formula = product_formula
+        self._quantum_instance = None
+        self._circuit_sampler = None
+        if quantum_instance is not None:
+            self.quantum_instance = quantum_instance
+        self._expectation = expectation
+
+    @property
+    def product_formula(self) -> ProductFormula:
+        """Returns a product formula used in the algorithm."""
+        return self._product_formula
+
+    @product_formula.setter
+    def product_formula(self, product_formula: ProductFormula) -> None:
+        """
+        Sets a product formula.
+        Args:
+            product_formula: A formula that defines the Trotterization algorithm.
+        """
+        self._product_formula = product_formula
+
+    @property
+    def quantum_instance(self) -> Union[QuantumInstance, Backend]:
```

Comment:
It's always converted to a quantum instance in the setter, and it can currently also be `None`
```suggestion
    def quantum_instance(self) -> Optional[QuantumInstance]:
```